ERCS/erc-7730.md

```diff
@@ -0,0 +1,1033 @@
+* The `intent` key contains a human readable string that wallets SHOULD display to explain to the user the intent of the function call. 
+* The `fields` key contains all the parameters that CAN be displayed, and the way to format them 
+  * The `required` key indicates which parameters wallets SHOULD display). 
+  
+In this example, the `_to` parameter and the `_value` SHOULD both be displayed, one as an address replaceable by a trusted name (ENS or others), the other as an amount formatted using metadata of the target ERC-20 contract (USDT). 
+
+### Common concepts
+
+#### Key naming convention
+
+In all the specification, key names starting with `$` are *internal* and have no value beyond readability of the specification file itself. They should not be used in any function to build the UI to review structured data.
+
+#### Structured Data
+
+This specification intends to be extensible to describe the display formatting of any kind of *structured data*. 
+
+By *Structured data*, we target any data format that has:
+* A well defined *type* system; the data being described itself being of a specific type
+* A description of the type system, the *schema*, that should allow splitting the data into *fields*, each field clearly identified with a *path* that can be descrived as a string.
+
+Displaying structured data is often done by wallets to review its content before authorizing an action in the form of a *signature* over some serialization of the structured data. As such, the structured data is contained in a *container structure*:
+* Container structure has a well defined *signature* scheme (a serialization scheme, a hashing scheme, and signature algorithm).
+* The container structure does not necessarily follow the same type system as the structured data.
+* Wallets receive the full container structure and uses the signature scheme to generate a signature on the overall structure.
+
+```mermaid
+block-beta
+  columns 1
+  block:container
+    columns 3
+    title["Container Structure"] space:2
+    a["Data"]
+    struct["Structured Data"]
+    c["Data"]
+  end
+  blockArrowId4<["Wallet"]>(down)
+  block:signature
+    columns 3
+    space sig["Signature"] space
+  end
+  style title fill:transparent,stroke:transparent
+  style struct fill:#555
+```
+
+Current specification covers EVM smart contract calldata:
+* Defined in Solidity
+* The schema is the ABI (expected in json format when linked to))
+* The container structure is an EVM Transaction serialized in RLP encoding
+
+It also supports EIP-712 messages
+* Defined in EIP-712
+* The schema is extracted from the message itself, from the `types` definitions and the `primaryType` key.
+* An EIP-712 message is self contained, the signature is applied to the hashed message itself.
+  
+In the future, it could be extended to structured data like Meta Transaction in [EIP-2771](./eip-2771.md) or User Operations in [EIP-4337](./eip-4337.md).
```

Comment:
```suggestion
```

If you'd like to talk about design choices made for future extensibility, do so in the Rationale section.